lib/src/resolver/did_registry.rs

```diff
@@ -56,6 +57,11 @@ impl DidattributeChangedFilter {
 
     /// Get the name of the attribute as a string. non-UTF8 bytes will be replaced with the unicode replacement character, �.
     pub fn name_string_lossy(&self) -> String {
+        let is_cstr = self.name.iter().rev().any(|&c| c == 0);
+        if is_cstr {
+            let cstr = unsafe { CStr::from_ptr(self.name.as_ptr() as *const c_char) };
```

Comment:
Since we're checking for the nul in the line beforehand, WDYT of using [`from_bytes_until_nul()`](https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.from_bytes_until_nul) and then `.expect("Checked for nul")` on the error, since we know it definitely contains at least one nul. Minor improvement, but it means we don't have any `unsafe` blocks